From aae5a6e41a7b6cdb1f0c2ccb5a50e51c5d93d8f4 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sun, 21 Jul 2019 09:32:37 +0200
Subject: [PATCH 351/386] telephony: mvebu_phone: obtain controller index from
 DT

Recent DT binding update of the TDM controller, allow to
get the unique port ID. Parse this information in the
probe function and inform user with the debug print.
This change is required for the multi-controller support.

Change-Id: I50422a4132bb0705e188f1cb05a430ccd51c4e3e
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12959
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/telephony/mvebu_phone/mv_phone.h     | 1 +
 drivers/telephony/mvebu_phone/mv_phone_dev.c | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/drivers/telephony/mvebu_phone/mv_phone.h b/drivers/telephony/mvebu_phone/mv_phone.h
index a5d639f029ca..0172362b63f2 100644
--- a/drivers/telephony/mvebu_phone/mv_phone.h
+++ b/drivers/telephony/mvebu_phone/mv_phone.h
@@ -119,6 +119,7 @@ struct mv_phone_params {
 };
 
 struct mv_phone_dev {
+	u32 id;
 	void __iomem *tdm_base;
 	void __iomem *pll_base;
 	void __iomem *dco_div_reg;
diff --git a/drivers/telephony/mvebu_phone/mv_phone_dev.c b/drivers/telephony/mvebu_phone/mv_phone_dev.c
index 68e318884750..c33e7cb53224 100644
--- a/drivers/telephony/mvebu_phone/mv_phone_dev.c
+++ b/drivers/telephony/mvebu_phone/mv_phone_dev.c
@@ -898,6 +898,11 @@ static int mvebu_phone_probe(struct platform_device *pdev)
 
 	priv->np = np;
 
+	if (of_property_read_u32(np, "cell-index", &priv->id)) {
+		dev_dbg(&pdev->dev, "using old DT, assume device ID = 0\n");
+		priv->id = 0;
+	}
+
 	mem = platform_get_resource_byname(pdev, IORESOURCE_MEM, "tdm_regs");
 	priv->tdm_base = devm_ioremap_resource(&pdev->dev, mem);
 	if (IS_ERR(priv->tdm_base))
@@ -1009,6 +1014,8 @@ static int mvebu_phone_probe(struct platform_device *pdev)
 			     tdmmc_if_pcm_tx_process, 0);
 	}
 
+	dev_info(&pdev->dev, "Registered TDM controller #%d\n", priv->id);
+
 	spin_lock_init(&priv->lock);
 	priv->dev = &pdev->dev;
 
-- 
2.17.1

